### PR TITLE
Heroku deployable version of Listo  as per #7

### PR DESCRIPTION
**Pull request by bertonjulian**
This will allow us to create a demo website that can be linked to the readme to give users a better idea of the tool. It will also help others quickly deploy their own version of Listo to Heroku. Heroku have a free tier which has everything we need to host a basic demo of Listo. It also allows teams which would allow me to add you both to the account. The Disk database will be used and stored in ephemeral storage on the container. It will get wiped every time the "Dyno" is restarted. Which is often with the free tier. Perfect for our situation. The Trello boards will persist within the Test Trello account however.  

https://listo-demo.herokuapp.com/

ToDo
- [x] Create a test Trello account for this demo site as they are free also. Currently using my account.
- [x] Create a readme to help (us) and other people to deploy to Heroku. Making it easier to deploy the tool.
- [x] Deploy the listo docker container to Dockerhub (https://hub.docker.com/r/listoproject/listo).
- [x] Add a link to the demo site to the main readme.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -21,11 +17,4 @@ services:
...
     ports:
-      - 8000:8000
-    links:
-      - database
-
-  database:
-    image: amazon/dynamodb-local
-    ports:
-      - 8000
+      - 8000:8000
```
Comment (zsims) [inline docker-compose.yml:20]: Nice! Really shows how much simpler the file database support has made it

File: server/src/dynamodb.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export class Dynamo implements Repository {
   }
 
   public async init() {
-    if (process.env.CREATE_DYNAMO_TABLES) {
+    if (process.env.LISTO_DATABASE === 'Dynamo' && process.env.CREATE_DYNAMO_TABLES) {
```
Comment (zsims) [inline server/src/dynamodb.ts:24]: Do you need this check? it'll only get run if  `init` is called, which won't happen unless this is the desired database?

**Review by zsims — APPROVED** [review]
Looks amazing, nice work :-)

For Trello, I imagine publicly readable board but only the demo can write to it? How often should it reset?

File: server/src/dynamodb.ts (hunk 2 above)
Comment (bertonjulian) [reply server/src/dynamodb.ts:24]: This still needs to be here for when Listo is deployed to prod in AWS, as the table is setup via a separate config (CloudFormation) right? And we dont want the code setting it up for us. This is only for local dev using the dynamo local db.

**Comment by bertonjulian** [comment]
> Looks amazing, nice work :-)
> 
> For Trello, I imagine publicly readable board but only the demo can write to it? How often should it reset? 

The boards can stay private as users provide their Trello email address during the assessment so get added to the board so they can view it! Do you think thats enough? Not sure re reset. Happy to let the boards accumulate, we can always right a script that deletes all boards every month if needed?

File: server/src/dynamodb.ts (hunk 2 above)
Comment (zsims) [reply server/src/dynamodb.ts:24]: ahh right. Hopefully we can cull that down into a single option somehow

File: .github/workflows/prod.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Build image
+      run: docker build . --file Dockerfile --tag $IMAGE_NAME
+    - name: Log into registry
+      run: echo "${{ secrets.DOCKER_API_KEY }}" | docker login -u ${{ secrets.DOCKER_USERNAME }} --password-stdin
```
Comment (zsims) [inline .github/workflows/prod.yml]:
```suggestion
      run: docker login -u ${{ secrets.DOCKER_USERNAME }} --password-stdin <<< "${{ secrets.DOCKER_API_KEY }}"
```

Not at all important, but you really don't need `echo` in 99% of cases

File: .github/workflows/prod.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    - name: Push image
+      run: | 
+        # Strip git ref prefix from version
+        VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+        
+        # Strip "v" prefix from tag name
+        [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
+
+        # Use Docker `latest` tag convention
+        [ "$VERSION" == "master" ] && VERSION=latest
```
Comment (zsims) [inline .github/workflows/prod.yml]:
```suggestion
        [[ "$VERSION" == "master" ]] && VERSION=latest
```

File: .github/workflows/prod.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    steps:
+    - uses: actions/checkout@v1
+    - name: Build image
+      run: docker build . --file Dockerfile --tag $IMAGE_NAME
+    - name: Log into registry
+      run: echo "${{ secrets.DOCKER_API_KEY }}" | docker login -u ${{ secrets.DOCKER_USERNAME }} --password-stdin
+    - name: Push image
+      run: | 
+        # Strip git ref prefix from version
+        VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
```
Comment (zsims) [inline .github/workflows/prod.yml]: if this is just bash you can do:

```bash
if [[ "${{ github.ref }}" =~ ^refs/tags/v/(.*) ]] ; then
  VERSION="${BASH_REMATCH[1]}"
elif [[ "${{ github.ref }}" =~ ^refs/heads/master ]] ; then
  VERSION="latest"
else
  VERSION="unknown"
fi
```

Would suggest using `version/` or `v/` as a prefix for tags as it'll organise them nicely for git clients (and avoids any prefix who-har)

File: .github/workflows/prod.yml [hunk 6]
```diff
@@ -0,0 +1,54 @@
+name: Build Listo, Push to Dockerhub and Deploy to Heroku
+
+on: 
+  push:
+    branches:
+      - master
+    tags:
+      - v*
```
Comment (zsims) [inline .github/workflows/prod.yml]:
```suggestion
      - 'v/*'
```

As discussed below

**Review by zsims — APPROVED** [review]
Looks great to me. Only non-functional comments (bash foo and `v/` instead of `v` prefix)

File: .github/workflows/prod.yml (hunk 3 above)
Comment (mr-beerkiss) [reply .github/workflows/prod.yml]: I gotta remember that one for myself

**Review by mr-beerkiss — APPROVED** [review]
LGTM

File: .github/workflows/prod.yml (hunk 5 above)
Comment (bertonjulian) [reply .github/workflows/prod.yml]: That looks much cleaner :)

Comment (bertonjulian) [reply .github/workflows/prod.yml]: Much simpler, nice :)

**Pull request merged**